bootstrap/src/main/java/io/airlift/bootstrap/LifeCycleManager.java

```diff
@@ -133,17 +133,31 @@ public void stop()
 
         List<Object> reversedInstances = new ArrayList<>(managedInstances);
         Collections.reverse(reversedInstances);
+        LifeCycleStopException stopException = null;
 
         for (Object obj : reversedInstances) {
             log.debug("Stopping %s", obj.getClass().getName());
             LifeCycleMethods methods = methodsMap.get(obj.getClass());
             for (Method preDestroy : methods.methodsFor(PreDestroy.class)) {
                 log.debug("\t%s()", preDestroy.getName());
-                preDestroy.invoke(obj);
+                try {
+                    preDestroy.invoke(obj);
+                }
+                catch (Exception e) {
+                    Throwable t = (e instanceof InvocationTargetException) ? ((InvocationTargetException) e).getTargetException() : e;
```

Comment:
This can be simplified to
```java
Throwable t = (e instanceof InvocationTargetException) ? e.getCause() : e;
```
Per the documentation, `getCause()` is the preferred way.

Reply:
Easy fix, will change.